pyteal/ast/subroutine.py

```diff
@@ -461,53 +648,93 @@ def evaluateSubroutine(subroutine: SubroutineDefinition) -> SubroutineDeclaratio
+        not one of the cases in the previous three options
 
     Usage (A) "argumentVars" - Storing pre-placed stack variables into local scratch space:
         Type 1. (by-value) use ScratchVar.store() to pick the actual value into a local scratch space
         Type 2. (by-reference) ALSO use ScratchVar.store() to pick up from the stack
             NOTE: SubroutineCall.__teal__() has placed the _SLOT INDEX_ on the stack so this is stored into the local scratch space
         Type 3. (ABI) abi_value.stored_value.store() to pick from the stack
+        Type 4. (ABI-output-arg) it is not really used here, since it is only generated internal of the subroutine
 
     Usage (B) "loadedArgs" - Passing through to an invoked PyTEAL subroutine AST:
         Type 1. (by-value) use ScratchVar.load() to have an Expr that can be compiled in python by the PyTEAL subroutine
         Type 2. (by-reference) use a DynamicScratchVar as the user will have written the PyTEAL in a way that satisfies
             the ScratchVar API. I.e., the user will write `x.load()` and `x.store(val)` as opposed to just `x`.
         Type 3. (ABI) use abi_value itself after storing stack value into scratch space.
+        Type 4. (ABI-output-arg) generates a new instance of the ABI value,
+            and appends a return expression of stored value of the ABI keyword value.
     """
 
     def var_n_loaded(
         param: str,
-    ) -> Tuple[ScratchVar, Union[ScratchVar, abi.BaseType, Expr]]:
-        loaded: Union[ScratchVar, abi.BaseType, Expr]
-        argVar: ScratchVar
+    ) -> tuple[ScratchVar, ScratchVar | abi.BaseType | Expr]:
+        loaded_var: ScratchVar | abi.BaseType | Expr
+        argument_var: ScratchVar
 
         if param in subroutine.by_ref_args:
-            argVar = DynamicScratchVar(TealType.anytype)
-            loaded = argVar
+            argument_var = DynamicScratchVar(TealType.anytype)
+            loaded_var = argument_var
         elif param in subroutine.abi_args:
             internal_abi_var = subroutine.abi_args[param].new_instance()
-            argVar = internal_abi_var.stored_value
-            loaded = internal_abi_var
+            argument_var = internal_abi_var.stored_value
+            loaded_var = internal_abi_var
         else:
-            argVar = ScratchVar(TealType.anytype)
-            loaded = argVar.load()
+            argument_var = ScratchVar(TealType.anytype)
+            loaded_var = argument_var.load()
 
-        return argVar, loaded
+        return argument_var, loaded_var
+
+    if len(subroutine.output_kwarg) > 1:
+        raise TealInputError(
+            f"ABI keyword argument num: {len(subroutine.output_kwarg)}. "
+            f"Exceeding abi output keyword argument max number 1."
+        )
 
     args = subroutine.arguments()
-    argumentVars, loadedArgs = zip(*map(var_n_loaded, args)) if args else ([], [])
+    args = [arg for arg in args if arg not in subroutine.output_kwarg]
```

Comment:
It looks like the `SubroutineDefinition.arguments()` function already filters out the keyword arg, so this is unnecessary?

Reply:
yes, I believe this is a segment in an early implementation, going to take that out.

Good catch